corehq/apps/case_importer/do_import.py

```diff
@@ -38,19 +38,40 @@
 
 
 def do_import(spreadsheet, config, domain, task=None, record_form_callback=None):
-    importer = _TimedAndThrottledImporter(domain, config, task, record_form_callback)
-    return importer.do_import(spreadsheet)
+    has_domain_column = 'domain' in [c.lower() for c in spreadsheet.get_header_columns()]
+    if has_domain_column and DOMAIN_PERMISSIONS_MIRROR.enabled(domain):
+        mirror_domains = DomainPermissionsMirror.mirror_domains(domain)
+        sub_domains = set()
+        import_results = _ImportResults()
+        for row_num, row in enumerate(spreadsheet.iter_row_dicts(), start=1):
+            if row_num == 1:
+                continue  # skip first row (header row)
+            sheet_domain = row.get('domain')
+            if sheet_domain != domain and sheet_domain not in mirror_domains:
+                err = exceptions.CaseRowError(column_name='domain')
+                err.title = _('Invalid domain')
+                err.message = _('Following rows contain invalid value for domain column.')
+                import_results.add_error(row_num, err)
+            else:
+                sub_domains.add(sheet_domain)
+        for sub_domain in sub_domains:
+            importer = _TimedAndThrottledImporter(sub_domain, config, task, record_form_callback, import_results)
```

Comment:
Can `multi_domain` be an argument to the constructor instead of being set separately? There's a risk of developers forgetting to set it, especially since there's no default value.